workbench-core/swb-common-ui/src/components/FileUpload/FileUploadComponent.tsx

```diff
@@ -0,0 +1,194 @@
+  /**
+   * Adds the specified ID to the root element of the component.
+   */
+  id?: string;
+  /**
+   * The main label for the form field.
+   */
+  label?: React.ReactNode;
+  /**
+   * Use to allow the selection of multiple files for upload from the
+   * user's local drive. It uses tokens to display multiple files.
+   * Files can be removed individually.
+   */
+  multiple?: boolean;
+  /**
+   * Called when the user selects a file.
+   * The event detail contains the current value.
+   * Not cancellable.
+   */
+  onChange?: (event: CustomEvent<ChangeDetail>) => void;
+  /**
+   * Specifies the currently selected file(s).
+   */
+  value?: File | File[];
+}
+
+const FileUploadComponent = (
+  {
+    accept = 'text/plain',
+    ariaLabel,
+    ariaRequired,
+    buttonText,
+    description,
+    disabled,
+    errorText,
+    fileMetadata,
+    constraintText,
+    id,
+    label,
+    multiple = false,
+    onChange,
+    value
+  }: FileUploadProps,
+  ref: ForwardedRef<ButtonProps.Ref>
+): JSX.Element => {
+  const fileInput = useRef<HTMLInputElement>(null);
+
+  const handleButtonClick = (): void | null => fileInput.current && fileInput.current.click();
+
+  const handleChange = useCallback(
+    ({ target }: ChangeEvent<HTMLInputElement>) => {
+      let newValue = undefined;
+      if (target.files && target.files[0]) {
+        newValue = multiple
+          ? value instanceof Array
+            ? [...value, target.files[0]]
+            : [target.files[0]]
+          : target.files[0];
+      }
+      if (onChange) {
+        onChange(new CustomEvent<ChangeDetail>('change', { detail: { value: newValue } }));
+      }
+    },
+    [value, multiple, onChange]
+  );
+
+  const handleDismiss = useCallback(
+    ({ detail }: CustomEvent<DismissDetail>) => {
+      const { index, file } = detail;
+      let newValue = value;
+      // eslint-disable-next-line security/detect-object-injection
+      if (multiple && value instanceof Array && value[index]) {
+        newValue = value.filter((f, i) => f !== file && i !== index);
+      }
+      if (onChange) {
+        onChange(new CustomEvent<ChangeDetail>('change', { detail: { value: newValue } }));
+      }
+    },
+    [value, multiple, onChange]
+  );
+
+  const baseButtonText = useMemo((): React.ReactNode => {
+    return buttonText || `Choose file${multiple ? 's' : ''}`;
+  }, [multiple, buttonText]);
+
+  const selectedFiles = useMemo((): React.ReactNode => {
+    if (errorText || !value) {
+      return null;
+    }
+
+    if (!multiple && value instanceof File) {
+      return <SelectedFile file={value} metadata={fileMetadata} multiple={false} />;
+    }
+
+    if (multiple && value instanceof Array) {
+      return <SelectedFileList fileList={value} metadata={fileMetadata} onDismiss={handleDismiss} />;
+    }
+
+    return null;
+  }, [errorText, value, multiple, fileMetadata, handleDismiss]);
+
+  return (
+    <SpaceBetween size="xs">
+      <FormField
```

Comment:
nit: I 'd add a `data-testid` attribute to components user interacts with, so when we do e2e testing of this component is easier

Reply:
Done. Also I included id in data-testid component in case if we have several FileUploadComponents on the page